ironfish-cli/src/commands/service/estimate-fee-rates.ts

```diff
@@ -0,0 +1,104 @@
+import { WebApi } from '@ironfish/sdk'
+import { Flags } from '@oclif/core'
+import { IronfishCommand } from '../../command'
+import { RemoteFlags } from '../../flags'
+import { IronfishCliPKG } from '../../package'
+
+export default class EstimateFees extends IronfishCommand {
+  static hidden = true
+
+  static description = `
+     Meaures fee rate estimates and submits them to telemetry API
+   `
+
+  static flags = {
+    ...RemoteFlags,
+    endpoint: Flags.string({
+      char: 'e',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'API host to sync to',
+    }),
+    token: Flags.string({
+      char: 't',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'API host token to authenticate with',
+    }),
+    delay: Flags.integer({
+      char: 'd',
+      required: false,
+      default: 60000,
+      description: 'Delay (in ms) to wait before uploading fee rate estimates',
+    }),
+  }
+
+  async start(): Promise<void> {
+    const { flags } = await this.parse(EstimateFees)
+
+    const apiHost = (
+      flags.endpoint ||
+      process.env.IRONFISH_API_HOST ||
+      'https://api.ironfish.network'
+    ).trim()
+
+    const apiToken = (flags.token || process.env.IRONFISH_API_TOKEN || '').trim()
+
+    const api = new WebApi({ host: apiHost, token: apiToken })
+
+    let connected = false
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      connected = await this.sdk.client.tryConnect()
+
+      if (!connected) {
+        await PromiseUtils.sleep(1000)
+        continue
+      }
+
+      const response = await this.sdk.client.estimateFeeRates({})
+
+      if (!(response.content.low && response.content.medium && response.content.high)) {
+        this.log('Unexpected response from fees/estimateFeeRates')
+      } else {
+        const feeRateLow = Number(CurrencyUtils.decode(response.content.low))
+        const feeRateMedium = Number(CurrencyUtils.decode(response.content.medium))
+        const feeRateHigh = Number(CurrencyUtils.decode(response.content.high))
+
+        await api.submitTelemetry({
+          points: [
+            {
+              measurement: 'fee_rate_estimate',
+              timestamp: new Date(),
+              fields: [
+                {
+                  name: `fee_rate_low`,
+                  type: 'integer',
+                  value: feeRateLow,
+                },
+                {
+                  name: `fee_rate_medium`,
+                  type: 'integer',
+                  value: feeRateMedium,
+                },
+                {
+                  name: `fee_rate_high`,
+                  type: 'integer',
+                  value: feeRateHigh,
+                },
+              ],
+              tags: [{ name: 'version', value: IronfishCliPKG.version }],
+            },
+          ],
+        })
+      }
+
```

Comment:
Is there a reason why you want to run this via CLI? I just recently used a pattern that sets up a graphile worker on the API to periodically run a job and submit metrics.
There are some changes mixed in for `minAndMaxDeposit` but should be simple to parse out. You might have a good reason for doing this way, just thought I would check :) 

https://github.com/iron-fish/ironfish-api/pull/1053/files

Reply:
We need to hit a running node's RPC endpoint for the fee estimates.

Now that we have authentication for RPC we could build out RPC calls from the API to a node, but I don't think we've used that pattern before. It seems like it could be lighter-weight than spinning up an EC2 instance to run this!

This will follow the pattern we've used for the syncer and deposit syncer -- a small EC2 instance runs a CLI command that pulls data from a node and sends it to the API.